agent/session/shell/shell_unix.go

```diff
@@ -105,16 +106,22 @@ func StartPty(
 		}
 
 		// Get the uid and gid of the runas user.
-		uid, gid, groups, err := getUserCredentials(log, sessionUser)
+		uid, gid, groups, homedir, err := getUserCredentials(log, sessionUser)
```

Comment:
a bit more work involved, but it would be nice if instead of just login every session user as the same `ssm-user`, a different linux server would be created for each IAM user.
that could be automated on first login.

ARN can be used to prevent duplicates if the bastion survives some IAM users